docs/source/federated-types/federated-directives.mdx

```diff
@@ -0,0 +1,240 @@
+---
+title: Federation-specific GraphQL directives
+---
+
+Apollo Federation defines a collection of **directives** that you use in your subgraph schemas to enable certain features.
+
+## Importing directives
+
+To use federated directives in a Federation 2 subgraph schema, that schema must include an `extend schema` definition with the following format:
+
+```graphql
+extend schema
+  @link(url: "https://specs.apollo.dev/federation/v2.0",
+        import: ["@key", "@shareable"])
+```
+
+The `import` array must include whichever federated directives your subgraph schema uses. The example above imports the [`@key`](#key) and [`@shareable`](#shareable) directives (which are used most commonly).
+
+### Renaming directives
+
+If a federated directive's name matches one of your own custom directives, you can rename the federated directive with the following syntax:
+
+```graphql
+extend schema
+  @link(url: "https://specs.apollo.dev/federation/v2.0",
+        import: [{ name: "@key", as: "@primaryKey"}, "@shareable"])
+```
+
+This example subgraph schema uses `@primaryKey` for the federated directive usually named [`@key`](#key).
+
+## `@key`
+
+Designates an object type as an [entity](../entities/).
+
+**Valid locations:** Object type definitions
+
+```graphql {1}
+type Product @key(fields: "id") {
+  id: ID!
+  name: String!
+  price: Int
+}
+```
+
+#### Arguments
+
+<table class="field-table">
+  <thead>
+    <tr>
+      <th>Name /<br/>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+
+<tbody>
+  
+<tr class="required">
+<td>
+
+##### `fields`
+
+`String`
+</td>
+<td>
+
+**Required.** A comma-separated list of field names that contribute to the entity's primary key.
```

Comment:
That's incorrect. Most importantly because it's *not* comma-separated. But there is also the fact that you  can have sub-selections.

Technically, `fields` is syntactically parsed like a graphQL [selection set](https://spec.graphql.org/draft/#sec-Selection-Sets) but without the top-level braces (even if in practice there is limitations to what is accepted), so that might the most concise-and-correct way to describe it.